apps/functions/applications-migration/folder-migration/src/folder-migration.js

```diff
@@ -0,0 +1,59 @@
+import { SynapseDB } from '../../common/synapse-db.js';
+import { makePostRequest } from '../../common/back-office-api-client.js';
+import { removeValues } from '../../common/utils.js';
+
+const query = 'SELECT * FROM [odw_curated_db].[dbo].[legacy_folder_data] WHERE caseReference = ?';
+
+/**
+ * Handle an HTTP trigger/request to run the migration
+ *
+ * @param {import('@azure/functions').Logger} logger
+ * @param {string[]} caseReferences
+ */
+export const migrateFolder = async (logger, caseReferences) => {
+	for (const caseReference of caseReferences) {
+		try {
+			logger.info(`reading Folders with caseReference ${caseReference}`);
+
+			const [folderRows, count] = await SynapseDB.query(query, {
+				replacements: [caseReference]
+			});
+
+			const folderEntities = folderRows.map((row) => {
+				const isRootFolder = row.parentFolderId === row.caseNodeId;
```

Comment:
`caseNodeId` isn't on the model. Does it need updating?

Reply:
> `caseNodeId` isn't on the model. Does it need updating?

I've asked ODW to set `parentFolderId` to `NULL` to identify a folder as being at the root level. So I'll remove this from our code now